swap_meet/vendor.py

```diff
@@ -1,2 +1,75 @@
 class Vendor:
-    pass
+    
+    def __init__(self, inventory=None): # Set a default value
+        if inventory is None: # Default value is falsy, so set to empty list
+            self.inventory = []
+        else:
+            self.inventory = inventory # If inventory is given, set it equal to inventory attribute
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+        else:
+            return False
+        return item
+
+    def get_by_category(self, category):
+        # Return a list of items that match category
+        # The vendor has an inventory list of items which have categories attached
+        items = [] # Creates empty list to return
+
+        for item in self.inventory: # For each item in inventory list
+            if item.category == category: # That item represents an Item instance, with a category attribute
+                items.append(item) # Append the whole item to the list
+
+        return items
+
+    def swap_items(self, person, my_item, their_item):
+        # Make self.inventory item replace itself with person.inventory item
+        if my_item in self.inventory and their_item in person.inventory:
+            self.remove(my_item)
+            self.add(their_item)
+
+            person.remove(their_item)
+            person.add(my_item)
+
+            return self.inventory
+        else:
+            return False
```

Comment:
As with `remove`, consider refactoring to first check for the edge case, and then move the main logic of the function out of the conditional. This is the design you use in the next function `swap_first_item`.

```suggestion
        if my_item not in self.inventory or their_item not in person.inventory:
            return None
            
        self.remove(my_item)
        self.add(their_item)

        person.remove(their_item)
        person.add(my_item)

        return self.inventory
```